File: online-judge-solutions/Codeforces/1682A.py
#https://codeforces.com/problemset/problem/1682/A
#Lang		:	Python 3.8
#Time		:	77 ms
#Memory         :	300 KB
t=int(input())
for x in range(t):
    n,s,d,c=int(input()),input(),0,''
    if(n%2!=0):
        d=1
        c=s[int(n/2)]
    else:c=s[int(n/2)-1]
    for i in range(int(n/2)-1,-1,-1):
        if(s[i]==c and s[n-1-i]==c):d+=2
        else:break
    print(d)

File: online-judge-solutions/Codeforces/1682A.cpp
//https://codeforces.com/problemset/problem/1682/A
//Lang		:	GNU C++14
//Time		:	46 ms
//Memory	:	300 KB
#include<iostream>
#include<algorithm>
using namespace std;
int main(){
	int t,n;
	cin>>t;
	while(t--){
		string s;
		cin>>n>>s;
		int d=0;
		char c;
		if(n%2!=0){d=1;c=s[n/2];}
		else{c=s[(n/2)-1];}
		for(int i=(n/2)-1;i>=0;i--){
			if(s[i]==c && s[n-1-i]==c) d+=2;
			else break;
		}
		cout<<d<<endl;
	}
	return 0;
}
